### ISSUE TITLE
Prevent the problem that the party data can not be read by the specification change

### ISSUE THREAD

**Issue by kagekiyo7** [MotocalDevelopers/motocal#154]
仕様変更で編成データが読み込めなくなる不具合を防ぐ

現在、スキルの第3スロットを実装した影響で過去の編成データが全て読み込めなくなっています。これからも仕様変更の度にこのようなことは起こりますし、その度にブラウザやサーバーに保存してある編成データが無駄になってしまいますので、編成データのロードの仕様を変えて防げないかIssuesを送らさせて頂きました。

Groovmanさんのコメント
>I look briefly at the save a load functions and the problem is mainly with how the save and load functions work.
>
>Disclaimer massive nerding on how saving and loading data works :)
>Saving serializes the current state of the app into a Json, loading deserializes the saved json and replaces the current state. When I add fields to weapons, or kei-gbf adds a new field for defense debuffs, how the state looks changes and when a state before those changes gets loaded the the newly added values are missing aka are undefined. That makes the app go dead, you could add a bunch of checks for undefined but that would just cause extra work. A proper solution is to write a function that resets the state basically an initializon function for the app does the trick can be used for both loading the app for the first time and resetting the app on load. Then you need to iterate over the saved data and assign values to the reset state that are in the saved state and in the app state and do the same for character objects and weapon objects by creating a new object and filling it and then adding it to the app state. So basically the load function need to be made a little bit more intelligent than just saving and overwriting the current state.
>
>Tl;dr: when loading reset the current state of the app and load the data into the clean state by iterating over the saved data instead of replacing the current state with the saved data.
>
>Hope it is understandable what I mean. Writing on my phone on the go so can’t proofread much :)

initialize関数を使って編成データ読み込み時にアプリの状態をリセットし、保存されたデータを繰り返し処理をすることで編成データの有効な値だけが代入された正しい状態で読み込める(?)。

**Comment by kei-gbf** [comment]
試験的ですがパッチあげてみます。#170

上で提案があるのは、未定義の参照が発生しないようにすべての項目を初期化しよう
というものですが、#170 は武器スキルとサポアビのみの部分的な対応です。

----

## 最近あった編成データの変更点

- [x] skill3
  - [x] skillname の変更 #138
- [x] support3
- [x] defenseDebuff
